spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,335 @@ def set_tabs_test(option)
+        core.cmd_sessions("--search", "last_checkin:after:200s last_checkin:before:30s")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "When the user specifies a before time that is larger than the after time" do
+        core.cmd_sessions("--search", "last_checkin:before:200s last_checkin:after:30s")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          After value must be a larger duration than the before value.
+        TABLE
+      end
+
+      it "When the user uses two before arguments with last checkin" do
+        core.cmd_sessions("--search", "last_checkin:before:200s last_checkin:before:30s")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          Cannot search for last_checkin with two before arguments.
+        TABLE
+      end
+    end
+
+    context "Searches with sessions that have different checkins and types" do
+      before(:all) do
+        Timecop.freeze(Time.parse('Dec 18, 2022 12:33:40.000000000 GMT'))
+      end
+
+      after(:all) do
+        Timecop.return
+      end
+
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "session1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: (Time.now - 90), type: 'java', sid:2, sname: "session2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          3 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: (Time.now - 20000), type: 'cmd_shell', sid:3, sname: "session3", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
+        }
+      end
+
+      it "When the user specifies both type and checkin" do
+        core.cmd_sessions("--search", "last_checkin:after:1m40.5s session_type:meterpreter")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            1   session1  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "When the user specifies both type and checkin but there are only partial matches" do
+        core.cmd_sessions("--search", "last_checkin:after:1m40.5s session_type:something")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context "searches for checkin with sessions that do not respond to checkin" do
```

Comment:
what do you think about adding a test to verify the session doesn't respond to checkin? that way if commandshell gets that feature or some future traveller updates this test unknowingly to a session that does respond to checkin it'll warn them